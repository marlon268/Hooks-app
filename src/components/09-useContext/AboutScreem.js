import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreem = () => {
   const { user, setUser } = useContext(UserContext);

   const handleClick = () => {
      setUser({});
   };

   return (
      <div>
         <h1>AboutScreem</h1>
         <hr />

         <pre>{JSON.stringify(user, null, 3)}</pre>
         <button className="btn btn-warning" onClick={handleClick}>
            Logout
         </button>
      </div>
   );
};
