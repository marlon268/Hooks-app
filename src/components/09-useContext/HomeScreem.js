import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreem = () => {

   const { user } = useContext(UserContext);


   return (
      <div>
         <h1>HomeScreem</h1>
         <hr />

         <pre>{JSON.stringify(user, null, 3)}</pre>
      </div>
   );

};
