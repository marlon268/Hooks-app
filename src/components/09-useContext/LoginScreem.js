import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreem = () => {
   const { setUser } = useContext(UserContext);

   return (
      <div>
         <h1>LoginScreem</h1>
         <hr />
         <button
            className="btn btn-dark"
            onClick={() =>
               setUser({
                  id: 123,
                  name: "Marlon",
                  email: "marlon@gmail.com",
               })
            }
         >
            Login
         </button>
      </div>
   );
};
