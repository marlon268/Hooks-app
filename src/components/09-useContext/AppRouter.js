import React from "react";
import { BrowserRouter as Router, Switch, Route, Redirect } from "react-router-dom";
import { AboutScreem } from "./AboutScreem";
import { HomeScreem } from "./HomeScreem";
import { LoginScreem } from "./LoginScreem";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
   return (
      <Router>
         <div>
            <NavBar />
            <div className="container">
               <Switch>
                  <Route exact path="/" component={HomeScreem} />
                  <Route exact path="/about" component={AboutScreem} />
                  <Route exact path="/login" component={LoginScreem} />
                  <Redirect to="/" />
               </Switch>
            </div>
         </div>
      </Router>
   );
};
