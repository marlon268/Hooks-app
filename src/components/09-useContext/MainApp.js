import React, { useEffect, useState } from "react";
import { AppRouter } from "./AppRouter";
import { UserContext } from "./UserContext";

const init = () => {
   return JSON.parse(localStorage.getItem("users")) || {};
};

export const MainApp = () => {
   const [user, setUser] = useState(init());

   useEffect(() => {
      localStorage.setItem("users", JSON.stringify(user));
   }, [user]);

   return (
      <UserContext.Provider
         value={{
            user,
            setUser,
         }}
      >
         <AppRouter />
      </UserContext.Provider>
   );
};
