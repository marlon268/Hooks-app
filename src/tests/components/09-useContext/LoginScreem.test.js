import React from 'react'
import { mount } from 'enzyme'
import { LoginScreem } from '../../../components/09-useContext/LoginScreem'
import { UserContext } from '../../../components/09-useContext/UserContext'


describe('Pruebas en <LoginScreem/>', () => {

   const setUser = jest.fn()

   const wrapper = mount(
      <UserContext.Provider value={{ setUser }}>
         <LoginScreem />
      </UserContext.Provider>
   )

   test('Debe de mostrarse correctamente', () => {
      expect(wrapper).toMatchSnapshot()
   })

   test('Debe de ejecutar el setUser con el argumento esperado ', () => {

      wrapper.find('button').prop("onClick")()

      expect(setUser).toHaveBeenCalledWith({
         id: 123,
         name: "Marlon",
         email: "marlon@gmail.com",
      })

   })


})
