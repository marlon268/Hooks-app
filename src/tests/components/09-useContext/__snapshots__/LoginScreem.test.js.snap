// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreem/> Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    LoginScreem
  </h1>
  <hr />
  <button
    className="btn btn-dark"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
